File: nodejs/fucking_coffee.js
#!/usr/bin/env node

/* Before running:
        npm install telnet-client
*/

var exec = require('child_process').exec;
var telnet = require('telnet-client');

var me = process.env['USERNAME'];

exec("who", function(error, stdout, stderr) {

    // Exit if no sessions with my username are found
    if(stdout.indexOf(me) == -1)
        process.exit(/*1*/);

    var coffee_machine_ip = process.env['COFFEE_MACHINE_IP'];
    var password = process.env['PASSWORD'];
    var con = new telnet();

    con.on('ready', function(prompt) {
        con.exec('Password: ' + password, function(error, res) {

            // Brew Coffee!
            con.exec('sys brew', function(error, res) {

                // Wait for 24s
                setTimeout(function() {

                    // Pour Coffee!
                    con.exec('sys pour', function(error, res) {
                        con.end();
                    });
                }, 24000);
            });
        });
    });

    con.connect({host: coffee_machine_ip});
});

File: rust/fucking_coffee.rs
use std::{
    io::{Read, Write},
    net::TcpStream,
    process::Command,
    thread,
    time::Duration,
};

const COFFEE_MACHINE_IP: &str = "10.10.42.42:23";
const COFFEE_MACHINE_PASSWORD: &str = "1234";

fn main() -> std::io::Result<()> {
    //Exit early if no sessions with my username are found
    let users = Command::new("who").output().unwrap();
    let is_active = String::from_utf8_lossy(&users.stdout)
        .split('\n')
        .any(|line| line.starts_with("username"));
    if !is_active {
        std::process::exit(0);
    }

    //Brew fucking coffee
    let delay_before_brew = Duration::from_secs(17);
    let delay = Duration::from_secs(24);

    thread::sleep(delay_before_brew);
    let mut conn = TcpStream::connect(COFFEE_MACHINE_IP)?;

    conn.write_all(COFFEE_MACHINE_PASSWORD.as_bytes())?;
    conn.read_exact(&mut [0u8; 1024])?;
    conn.write_all(b"sys brew")?;

    thread::sleep(delay);
    conn.write_all(b"sys pour")?;

    Ok(())
}
